### PR TITLE
Deps: Update core to v0.6

### PR DESCRIPTION
**Pull request by TheCharlatan**
This includes a massive rename of `Offer` into `Deal`.
s/PublicOffer/Deal/
s/PublicOffer.Offer/Deal.Parameters/

Also, the swap id and the deal id now have the same underlying type, allowing us to track the entire lifetime of a swap from a made deal to a swap outcome.

### CODE REVIEW

File: src/bus/info.rs [hunk 1]
```diff
@@ -212,41 +212,25 @@ pub struct MoneroAddressSwapIdPair {
...
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[display(TookOffer::to_yaml_string)]
-pub struct TookOffer {
-    pub offerid: Uuid,
+#[display(TookDeal::to_yaml_string)]
+pub struct TookDeal {
+    pub dealid: DealId,
```
Comment (h4sh3d) [inline src/bus/info.rs:230]: Maybe `deal_id`?

File: src/bus/p2p.rs [hunk 2]
```diff
@@ -169,7 +169,7 @@ impl Reveal {
 #[display("{commit}")]
 pub struct TakerCommit {
     pub commit: Commit,
-    pub public_offer: PublicOffer, // TODO: replace by public offer id
+    pub deal: Deal, // TODO: replace by public deal id
```
Comment (h4sh3d) [inline src/bus/p2p.rs:172]: Comment out-of-date, either
1. we do now the switch to `DealId`
2. update the comment for later
3. we consider that using plain `Deal` is ok, drop the comment

I'd do 1 or 3 personally

File: src/cli/opts.rs [hunk 3]
```diff
@@ -58,24 +58,24 @@ pub enum Command {
     #[clap(aliases = &["ls"])]
     ListSwaps,
 
-    /// Lists public offers created by daemon
+    /// Lists public deals created by daemon
     #[clap(aliases = &["lo"])]
```
Comment (h4sh3d) [inline src/cli/opts.rs:62]: Change to `ld` aliases

File: src/cli/opts.rs [hunk 4]
```diff
@@ -58,24 +58,24 @@ pub enum Command {
...
             default_value = "open",
             possible_values = &["open", "Open", "inprogress", "in_progress", "ended", "Ended", "all", "All"],
         )]
-        select: OfferSelector,
+        select: DealSelector,
     },
 
-    /// Gives information on an open offer
+    /// Gives information on an open deal
     #[clap(aliases = &["oi"])]
```
Comment (h4sh3d) [inline src/cli/opts.rs:74]: Change to `di` aliases

File: src/cli/opts.rs [hunk 5]
```diff
@@ -221,20 +221,20 @@ pub enum Command {
         #[clap(long = "xmr-addr")]
         monero_address: XmrAddress,
 
-        /// An encoded public offer.
-        #[clap(short = 'o', long = "offer")]
-        public_offer: PublicOffer,
+        /// An encoded public deal.
+        #[clap(short = 'o', long = "deal")]
```
Comment (h4sh3d) [inline src/cli/opts.rs:225]: I proposed in the wiki to use short `-D`

File: src/cli/opts.rs [hunk 6]
```diff
@@ -221,20 +221,20 @@ pub enum Command {
...
         #[clap(short, long)]
         without_validation: bool,
     },
 
-    /// Revoke offer accepts an offer and revokes it within the runtime.
-    #[display("revoke-offer<{public_offer}>")]
-    RevokeOffer {
-        /// The offer to be canceled.
-        public_offer: PublicOffer,
+    /// Revoke deal accepts an deal and revokes it within the runtime.
```
Comment (h4sh3d) [inline src/cli/opts.rs:233]: Maybe do a global search `s/an deal/a deal/`

File: src/farcasterd/trade_state_machine.rs [hunk 7]
```diff
@@ -36,10 +36,10 @@ use std::str::FromStr;
 ///       |            |         ____|____
 ///       |            |        |         |
 ///       |            V        |         V
-///       |        MakeOffer    |   TakerConnect
+///       |        MakeDeal    |   TakerConnect
```
Comment (h4sh3d) [inline src/farcasterd/trade_state_machine.rs:39]: Shifted alignment

File: src/farcasterd/trade_state_machine.rs [hunk 8]
```diff
@@ -919,27 +905,27 @@ fn attempt_transition_to_take_offer(
...
+        deal,
         arb_addr,
         acc_addr,
         peerd,
-    } = take_offer;
+    } = take_deal;
     match event.request {
         BusMsg::Ctl(CtlMsg::SwapKeys(swap_keys)) => {
-            let swap_id: SwapId = SwapId::random();
+            let swap_id: SwapId = deal.id().into(); // The deal id is now used to track a swap
```
Comment (h4sh3d) [inline src/farcasterd/trade_state_machine.rs:921]: :tada:

File: src/grpcd/runtime.rs [hunk 9]
```diff
@@ -1393,14 +1382,17 @@ impl Farcaster for FarcasterService {
             .map_err(|_| Status::invalid_argument("arbitrating address"))?;
         let monero_address = monero::Address::from_str(&str_monero_address)
             .map_err(|_| Status::invalid_argument("accordant_address"))?;
-        let public_offer = PublicOffer::from_str(&str_public_offer)
-            .map_err(|_| Status::invalid_argument("public offer"))?;
+        let deal =
+            Deal::from_str(&str_deal).map_err(|_| Status::invalid_argument("public deal"))?;
```
Comment (h4sh3d) [inline src/grpcd/runtime.rs:1386]: Global search for `s/public deal/deal/`

File: src/bus/info.rs (hunk 1 above)
Comment (TheCharlatan) [reply src/bus/info.rs]: fbc278d0575c8a5d8dc4513b935801b5ba09f4d4

File: src/bus/p2p.rs (hunk 2 above)
Comment (TheCharlatan) [reply src/bus/p2p.rs]: 375046e93ac17e3d1ec54309a1f41d04dacdb0fc

File: src/cli/opts.rs (hunk 3 above)
Comment (TheCharlatan) [reply src/cli/opts.rs]: 64b2f5c7fd5ce32f6b166ed1a6f8f9dfb62f2606

File: src/cli/opts.rs (hunk 4 above)
Comment (TheCharlatan) [reply src/cli/opts.rs]: 64b2f5c7fd5ce32f6b166ed1a6f8f9dfb62f2606

File: src/cli/opts.rs (hunk 5 above)
Comment (TheCharlatan) [reply src/cli/opts.rs]: 916c4d5599a56c85111433daafe141f78906f882 :P

File: src/cli/opts.rs (hunk 6 above)
Comment (TheCharlatan) [reply src/cli/opts.rs]: 39ac90310b0f578fda9a37fd9da84051e0e9d93f

File: src/farcasterd/trade_state_machine.rs (hunk 7 above)
Comment (TheCharlatan) [reply src/farcasterd/trade_state_machine.rs]: d0e899f3885a52cce37d2a8ced56cfc4dabe40fb

File: src/grpcd/runtime.rs (hunk 9 above)
Comment (TheCharlatan) [reply src/grpcd/runtime.rs]: dc5fe2b699596883ce4d3b775656e20f604299ca

**Review by h4sh3d — APPROVED** [review]

**Pull request merged**